plugin-core/src/main/java/appland/telemetry/TelemetryProperties.java

```diff
@@ -0,0 +1,62 @@
+package appland.telemetry;
+
+import appland.AppMapPlugin;
+import appland.utils.GsonUtils;
+import com.google.gson.annotations.SerializedName;
+import com.intellij.openapi.application.ApplicationInfo;
+import com.intellij.openapi.util.SystemInfo;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class TelemetryProperties {
+    public static Map<String, String> getProperties(boolean includeUsername) {
+        var properties = new HashMap<String, String>();
+        properties.put("extname", AppMapPlugin.getDescriptor().getPluginId().getIdString());
+        properties.put("extversion", AppMapPlugin.getDescriptor().getVersion());
+        properties.put("ide", ApplicationInfo.getInstance().getFullApplicationName());
+        properties.put("ideversion", ApplicationInfo.getInstance().getFullVersion());
+        properties.put("os", SystemInfo.OS_NAME);
+        properties.put("osversion", SystemInfo.OS_VERSION);
+        properties.put("jvmversion", System.getProperty("java.version"));
+        properties.put("product", ApplicationInfo.getInstance().getBuild().getProductCode());
+        properties.put("source", "JetBrains");
+
+        if (includeUsername) {
+            try {
+                properties.put("username", System.getProperty("user.name"));
+            } catch (Exception e) {
+                // ignore
+            }
+        }
+        return properties;
+    }
+
+    public static @NotNull String getPropertiesAsJson() {
```

Comment:
IMHO the name should make it clear that it's the CLI properties and not all properties.